frontend/lib/src/AppNode.ts

```diff
@@ -707,12 +719,14 @@ export class AppRoot {
       }
 
       case "addBlock": {
+        const deltaMessageId = Date.now()
```

Comment:
Does date really matter? Would it make sense just to use an uuid here?

Reply:
yeah it could also be a UUID! The field was originally called `receivedAt`, which is how the timestamp was introduced. Since uniqueness across elements does not matter, I kept it. But I am happy to use a UUID or at least I can leave a comment along the lines `// using timestamp here to indicate when the message was received; uniqueness across elements is not necessary`
Do you have a strong preference for one of the two options?

Reply:
Opinion is not strong.

I am perhaps missing the understanding, is this updating practically every script run with the updated date? Wouldn't this then be equivalent to just running the effect every time?

Reply:
No, it's set only once when the delta message is received. This will allow us to trigger a React re-render of a component when a new delta message was received from the backend, even though no props changed; in this case  React would skip the re-render - e.g. `delta.element.is_open` has been `true` all the time - which is problematic for elements / blocks that have UI-only states.
In this concrete case, dismissing a dialog is a UI-only event, so when I click the button to open the dialog, the backend sends following message: `{"title": "Some dialog", "isOpen": true}`
Now I dismiss it, click on the button again and the backend sends the same message again: `{"title": "Some dialog", "isOpen": true}`
For React, no props changed so the `useEffect` hook is not re-run and the dialogs stays closed.
With this id, we basically can manually tell React to rerun the hook whenever the dialog message was received.

However, the id is only set once when receiving the delta and stays stable for an element, because if it would refresh with every render, the dialog would be opened even if I interact with a completely unrelated fragment. But now, when I interact with an unrelated fragment, no new delta message for the dialog is sent and, thus, the `deltaMessageId` is _not_ updated (so the hook is not running every time). This is why I extended the `e2e_playwright/st_dialog_test.py` with this scenario.

Sorry, I have the feeling my thought process write-up here has room for improvement 🙂 
But does it make more sense now? If not I will try to think of a better way to express my thinking!